tree/ntupleutil/v7/src/RNTupleInspector.cxx

```diff
@@ -228,42 +229,62 @@ const std::vector<ROOT::ENTupleColumnType> ROOT::Experimental::RNTupleInspector:
 void ROOT::Experimental::RNTupleInspector::PrintColumnTypeInfo(ENTupleInspectorPrintFormat format, std::ostream &output)
 {
    struct ColumnTypeInfo {
-      std::uint32_t count;
-      std::uint64_t nElems, compressedSize, uncompressedSize;
+      std::uint64_t nElems = 0;
+      std::uint64_t compressedSize = 0;
+      std::uint64_t uncompressedSize = 0;
+      std::uint64_t nPages = 0;
+      std::uint32_t count = 0;
 
       void operator+=(const RColumnInspector &colInfo)
       {
          this->count++;
          this->nElems += colInfo.GetNElements();
          this->compressedSize += colInfo.GetCompressedSize();
          this->uncompressedSize += colInfo.GetUncompressedSize();
+         this->nPages += colInfo.GetNPages();
+      }
+
+      // Helper method to calculate compression factor
+      float GetCompressionFactor() const
+      {
+         if (uncompressedSize == 0)
+            return 1.0;
+         return static_cast<float>(uncompressedSize) / static_cast<float>(compressedSize);
       }
    };
 
-   std::map<ROOT::ENTupleColumnType, ColumnTypeInfo> colTypeInfo;
+   std::map<ENTupleColumnType, ColumnTypeInfo> colTypeInfo;
 
+   // Collect information for each column
    for (const auto &[colId, colInfo] : fColumnInfo) {
       colTypeInfo[colInfo.GetType()] += colInfo;
    }
 
    switch (format) {
    case ENTupleInspectorPrintFormat::kTable:
-      output << " column type    | count   | # elements      | compressed bytes  | uncompressed bytes\n"
-             << "----------------|---------|-----------------|-------------------|--------------------" << std::endl;
-      for (const auto &[colType, typeInfo] : colTypeInfo) {
+      // table header with compression ratio and pages
```

Comment:
The header contains more information that this. IMO it's also obvious from the code itself what it is, so I would remove it altogether.